app/src/org/commcare/logging/DataChangeLogger.kt

```diff
@@ -0,0 +1,155 @@
+package org.commcare.logging
+
+import android.content.Context
+import android.os.Environment
+import android.util.Log
+import org.apache.commons.lang3.StringUtils
+import org.commcare.android.logging.ReportingUtils
+import org.commcare.utils.CrashUtil
+import org.commcare.utils.FileUtil
+import org.javarosa.core.io.StreamsUtil
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.IOException
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ *  Used to log some crtitical platform events like DB migrations, Clear user data,
+ *  app installations etc. All these logs are backed by file storage and are accessible
+ *  from outside CommCare
+ */
+class DataChangeLogger {
+
+    fun init(context: Context) {
+        initLogFiles(context)
+    }
+
+    private fun initLogFiles(context: Context) {
+        if (!isExternalStorageWritable()) {
+            Log.e(TAG, "External Storage unavialable to write logs")
+            return
+        }
+
+        primaryFile = initFile(context, PRIMARY_LOG_FILE_NAME)
+        if (primaryFile!!.exists() && primaryFile!!.length() > MAX_FILE_SIZE) {
+            // When primary file gets full -> Delete secondary and move logs in primary to secondary
```

Comment:
Were you able to test this behavior manually? (the easiest way is probably to reduce the max file size)

Reply:
Yup. It works as expected.